src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,164 @@
+    private static final String LOGIN3 = "login3";
+    private static final String PASSWORD1 = "password1";
+    private static final String PASSWORD2 = "password2";
+    private static final String PASSWORD3 = "password3";
+    private static final int AGE1 = 22;
+    private static final int AGE2 = 40;
+    private static final int AGE3 = 18;
+    private static final int SMALL_AGE = 12;
+    private static final int UNREAL_AGE = -10;
+    private static final long USER_ID1 = 1L;
+    private static final long USER_ID2 = 2L;
+    private static final long USER_ID3 = 3L;
+    private static RegistrationService registrationService;
+    private static StorageDao storage;
+    private User user1;
+    private User user2;
+    private User user3;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storage = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user1 = new User(USER_ID1,LOGIN1, PASSWORD1, AGE1);
+        user2 = new User(USER_ID2,LOGIN2, PASSWORD2, AGE2);
+        user3 = new User(USER_ID3,LOGIN3, PASSWORD3, AGE3);
+    }
+
+    @Test
+    void registerAndGetOrdinaryUsers() {
+        registrationService.register(user1);
+        User actual = storage.get(LOGIN1);
+        assertEquals(user1,actual);
+
+        registrationService.register(user2);
+        actual = storage.get(LOGIN2);
+        assertEquals(user2,actual);
+
+        registrationService.register(user3);
+        actual = storage.get(LOGIN3);
+        assertEquals(user3,actual);
+    }
+
+    @Test
+    void getUserByUnrealLogin() {
+        assertEquals(NULL_VALUE, storage.get(UNREAL_LOGIN1));
+
+        assertEquals(NULL_VALUE, storage.get(UNREAL_LOGIN2));
+    }
+
+    @Test
+    void registerAndGetUnrealUsers() {
+        User actual = storage.get(LOGIN1);
+        assertNotEquals(user2,actual);
+
+        actual = storage.get(LOGIN2);
+        assertNotEquals(user3,actual);
+
+        actual = storage.get(LOGIN3);
+        assertNotEquals(user1,actual);
+    }
+
+    @Test
+    void registerNullUser() {
+        User userNull = (User) NULL_VALUE;
+        Assertions.assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userNull);
+        },"Expected " + UserIsNotValidException.class.getName()
+                + " to be thrown for the null user, but it wasn't");
+    }
+
+    @Test
+    void registerUserWithNullLogin() {
+        User userWithLogin = new User(USER_ID1, (String) NULL_VALUE, PASSWORD, AGE2);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithLogin);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the null user login, but it wasn't");
+    }
+
+    @Test
+    void registerUserWithNullPassword() {
+        User userWithPassword = new User(USER_ID1, LOGIN, (String) NULL_VALUE, AGE1);
```

Comment:
you have set up users in the beginning, use and modify them if you need instead of creating a new one each time, same for other tests